backend/README.md

```diff
@@ -0,0 +1,103 @@
+# Setup Guide
+
+## Step 1: Set up the server
+
+1. **Open Command Prompt:**
+   Open the command prompt on your machine.
+
+2. **Navigate to the project directory:**
+   Go to the server folder of your project:
+
+   ```bash
+   cd <your-project-path>/backend
+
+   ```
+
+3. **Install Dependencies:**
+   Run the following command to install the required dependencies:
+
+   ```bash
+   npm install --force
+   ```
+
+4. **Set up environment variables:**
+   - Copy the content from the `.env.example` file
+   - Create a new `.env` file in the same location and paste the copied content.
+   - Update the `.env` file as per your configuration needs.
+5. **Run the development server:**
+   Start the server by running:
+
+   ```bash
+   npm run dev
+   ```
+
+6. **Check if the server is running:**
+   Open your browser and visit the following URL:
+
+   ```bash
+   http://localhost:8080/ping
+   ```
+
+   If you get a pong response, your server is running correctly.
+
+## Step 2: Set up Prometheus
+
+1. **Configure Prometheus:**
+
+   - Open the `prometheus-config.yml` file.
+   - In the `targets` array, add the IP address of your local machine
+     - If you're running this on a production server, replace it with the production server's IP address.
```

Comment:
@tejasbenibagde here you mention If I have to run this on production I need to add the IP address of deployed server in Targets but it is not safe to directly add the IP address to the `prometheus-config.yml` as it will be visible. Can we use env variables for this in Targets to substitute IP address. If so let me know how?

Reply:
I added a new way of doing it. I added the `prometheus-config.example.yml` file. made sure that eaveryone can rename the file to `prometheus-config.yml` and add their ip addresses their freely.  I also added  `prometheus-config.yml`  file to .gitignore to make sure that it does not get in the accidental git pushes. I updated `README` according to it as well.

Reply:
@tejasbenibagde that's fine, but my concern is if I want to run Prometheus for my deployed server then how can I give the IP address of that deployed server in the Prometheus config file without disclosing the server IP address in file. Since this config file needs to be added in repo in ordered to be deployed on server.
Since I want the monitoring of my deployed server and not the local instance.

Let me know if you get my query or not.

Reply:
@AlfiyaSiddique you can just change few things before the deployment like removing the prometheus-config.yml from the .gitignore file. and push it in production.  Isn't it simple ?

Reply:
No this is not possible @tejasbenibagde, the deployment is linked with Github so every time a new merge is made the deployment makes the current image of the code present on Github so this can't be done. Any other solution?

Reply:
If your concern is that it will break the code then be assured that it will not. Cause I made sure that the server runs the same way as it should be even if you fail to correctly configure the Prometheus and grafana. And as for the production. You will have to find your way through. Cause I neither have the idea of how your production setup and neither is my part of work sorry.